### ISSUE TITLE
Add option for scientific articles

### ISSUE THREAD

**Issue by tiotrom** [SimonBrazell/privacy-redirect#96]
What if you were to add something of the sort https://addons.mozilla.org/en-US/firefox/addon/sci-hub-now/?src=search :) ? 

"When current page contains DOI identifier of a restricted academic paper, find it on Sci-Hub with just a single click on toolbar icon or select your desired DOI, then right click to selected text to use Sci-Hub Now!"

To basically unlock scientific articles. If it helps, SearX uses such a  thing https://github.com/asciimoo/searx.

**Comment by SimonBrazell** [comment]
Yeah sounds cool, I love Sci-Hub, would you be able to send me an example of a page where you'd see something like this working with instructions etc?

**Comment by tiotrom** [comment]
I need to search for this, but as I can tell they use DOI - https://asciimoo.github.io/searx/admin/plugins.html?highlight=doi https://github.com/asciimoo/searx/issues?q=Open+Access+DOI . Links seem to go through this https://dissem.in/. And specifically for sci-hub they seem to use this https://github.com/zaytoun/scihub.py.

More importantly this is how Sci-Hub Now is doing it https://github.com/0x01h/sci-hub-now/ - maybe make it so that it is automatic. Like this one does https://github.com/patztablook22/sci-hub.

**Comment by tiotrom** [comment]
Heads Up, the original github link of the extension is not maintained anymore, but this is https://github.com/gchenfc/sci-hub-now - I would so love to see SciHub in your privacy redirect list. This one is great because users can change the sci-hub instance in their preferences so I'd imagine you can force that in the code!? Here are the sci-hub instances that work:

- https://sci-hub.st/
- https://sci-hub.se/
- https://scihubtw.tw/
- https://sci-hub.do/

**Comment by tiotrom** [comment]
Maybe this last pull from the developer of Sci-Hub Now https://github.com/gchenfc/sci-hub-now/pull/11 is helpful for this extension too.

**Comment by seth1** [comment]
One issue is sci-hub is not super legal in many countries and even blocked in others (Sweden off the top of my head).
This could change the extension to be viewed as something not only privacy freaks use but also criminals (might be an issue if you're at university perhaps). On the other hand, tor is already basically viewed as illegal in many countries (Iran I think?) and privacy redirect ships with onion instances so perhaps that doesn't really matter.

**Comment by tiotrom** [comment]
Oh I'm sure invidious and others are not "legal" from the standpoint of youtube or other platforms. Scihub is praised by so many scientists and researchers. We had Alexandra from Scihub to our podcast a few months ago https://www.tromsite.com/videos/#fvp_262,1s talking about scihub if you are interested. She does a fantastic job.

**Comment by SimonBrazell** [comment]
@tiotrom I just tried to watch your video but the server seems very slow, couldn't stream so I decided to download it instead, speed is sitting around 20KB/s with an ETA of 19 hours 🙁

**Comment by seth1** [comment]
> Oh I'm sure invidious and others are not "legal" from the standpoint of youtube or other platforms.

I'm afraid I do not know what you mean by this since you put legal in quotes. From what I understand invidious and the other instances are not illegal (with the exception of whatever.onion, if you live in Central Asia or something), even if individuals in youtube might wish it was, although maybe it violates a term of service somewhere? If it is I would like to see evidence so that I may better inform myself.

I myself quite like sci-hub and various related things, but I merely wanted to raise a point for consideration in the project. I'll be sure to listen to your podcast.

**Comment by tiotrom** [comment]
> @tiotrom I just tried to watch your video but the server seems very slow, couldn't stream so I decided to download it instead, speed is sitting around 20KB/s with an ETA of 19 hours slightly_frowning_face

Hm, it streams from archive.org. We have it on peertube too https://gorf.tube/videos/watch/playlist/284c2399-0a95-423d-b15e-158ddc126846?playlistPosition=57&resume=true - may I ask where you are trying to stream from? For me, from Spain, works perfectly fine. Maybe your region + internet connection are very slow?

**Comment by SimonBrazell** [comment]
I'm in Australia, the internet connection is fine, I can stream anything else, the peertube link works for example, just the original was terribly slow.

**Comment by tiotrom** [comment]
@seth1 I put legal in quotes because it depends what is legal here may not be there. And it is not like legal = fair anyways. But yah, scihub is such a wonderful resource used by hundreds of thousands of people daily. There are addons for firefox and chrome already, that redirect scientific articles to scihub, so probably should be ok.

**Comment by tiotrom** [comment]
> I'm in Australia, the internet connection is fine, I can stream anything else, the peertube link works for example, just the original was terribly slow.

Thanks for the info. We've heard in the past similar stories abotu our videos...we will look more into it.
